### ISSUE TITLE
Prioritize the migration of recent expensive to compute data during RM scale down

### ISSUE THREAD

**Issue by holdenk** [dask/distributed#4202]

**Comment by quasiben** [comment]
RM scale down is Resource Manager Scale down ?  For example, Kubernetes/Yarn/etc removing workers ?

**Comment by mrocklin** [comment]
Yes, but this would be implemented more generally.  I think that this is
handled in Scheduler.retire_workers which relies on Scheduler.rebalance
